# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Location.cpp
# Commit: ddf4012a7652
# Full Hash: ddf4012a7652e36d144fc43bc99335276deeafbe
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-05-22 03:40:58
# Regressor Bug: 965637
# File Overlap Count: 1
# Description:
#   Bug 965637: Move CSP from Principal into Client, part 1: backend changes. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27654
# ==============================================================================

diff -r 6c2047bc2f3e -r ddf4012a7652 dom/base/Location.cpp
--- a/dom/base/Location.cpp	Tue May 21 22:51:54 2019 +0000
+++ b/dom/base/Location.cpp	Tue May 21 23:14:27 2019 +0000
@@ -150,15 +150,9 @@
   RefPtr<nsDocShellLoadState> loadState = new nsDocShellLoadState(aURI);
 
   loadState->SetTriggeringPrincipal(triggeringPrincipal);
-
-  // Currently we query the CSP from the triggeringPrincipal, which is the
-  // doc->NodePrincipal() in case there is a doc. In that case we can query
-  // the CSP directly from the doc after Bug 965637. In case there is no doc,
-  // then we also do not need to query the CSP, because only documents can have
-  // a CSP attached.
-  nsCOMPtr<nsIContentSecurityPolicy> csp;
-  triggeringPrincipal->GetCsp(getter_AddRefs(csp));
-  loadState->SetCsp(csp);
+  if (doc) {
+    loadState->SetCsp(doc->GetCsp());
+  }
 
   if (sourceURI) {
     nsCOMPtr<nsIReferrerInfo> referrerInfo =